apps/sms/views/conversation/js/attachment.js

```diff
@@ -26,8 +26,7 @@
   function Attachment(blob, options) {
     options = options || {};
     this.blob = blob;
-    this.name = blob.name || options.name ||
-      navigator.mozL10n.get('unnamed-attachment');
+    this.name = blob.name || options.name || 'undefined';
```

Comment:
ugh. Let's handle this in a follow up patch. I'm not sure if I know how to implement your solution.
Also, I believe that it's a bad habit to store localized string in the data model, since it's a view role to display it.

So I'm more on the side of:
```
function Attachment(name) {
  this.name = name;
}
[...]

function render() {
  if (!attachment.name) {
    element.textContent = attachment.name;
  } else {
    element.setAttribute('data-l10n-id', 'unnamed-attachment');
  }
}
```

But since it seems complex enough, let's come back to it in a separate patch.